### PR TITLE
LESQ-732: Utilise new API on Unit Listing page

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 1960

- Updates Unit Listing page to point at new API
- Remove theme title from Unit cards
- Creates Unit Listing fixture data

## Issue(s)

Fixes #LESQ-732

## How to test

1. Go to https://deploy-preview-2356--oak-web-application.netlify.thenational.academy
2. Click on any unit listing page, both legacy and new content should display

## Screenshots

How it used to look (delete if n/a):
Screenshot 2024-04-12 at 10 55 54

How it should now look:
Screenshot 2024-04-12 at 10 55 18
Screenshot 2024-04-12 at 10 59 31

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
- Is it going to be really annoying if I ask for "Key Stage" (in the breadcrumb and by the unit title) to become "Key stage"? 
Screenshot 2024-04-15 at 09 52 48
- Legacy KS4 maths "core" isn't listed as one of the options in the tier selector on the unit listing page, similar thing has happened in search because "Core" doesn't exist in new content  
Screenshot 2024-04-15 at 09 57 59
- Legacy content doesn't have any learning theme filters (computing ks4 for example has some learning theme filters, but on this deploy they aren't there). Weirdly, the new content still does. I had assumed they would all have them now (not included on the unit listing themselves, just selectable on the right hand side

File: src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.query.ts [hunk 1]
```diff
@@ -24,7 +25,13 @@ const unitListingQuery =
       });
     }
 
-    return unitListingSchema.parse(programme);
+    return unitListingSchema.parse({
+      ...programme,
+      keyStageTitle:
+        programme.keyStageTitle &&
+        programme.keyStageTitle.charAt(0).toUpperCase() +
```
Comment (JBR90) [inline src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.query.ts:32]: We should add a helper functions for this, doesn't have to be this PR

**Review by JBR90 — COMMENTED** [review]
I think you need to merge in main to fix the missing blog error

File: src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.gql [hunk 2]
```diff
@@ -1,9 +1,6 @@
-query unitListing($programmeSlug: String!, $isLegacy: Boolean!) {
+query unitListing($programmeSlug: String!) {
   programme: published_mv_unit_listing_page_5_0_0(
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.gql:2]: do we not need to update to the latest MV to get core?

Comment (k-huggs) [reply src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.gql]: ✅

File: src/node-lib/curriculum-api-2023/helpers/index.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { TierSchema } from "../queries/unitListing/unitListing.schema";
+
+export const toSentenceCase = (str: string) => {
+  return str.charAt(0).toUpperCase() + str.slice(1).toLowerCase();
+};
+
+const hasCoreTierSet = new Set(["maths-ks4"]);
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/helpers/index.ts:7]: just noting that when we make the new MV we should look into refactoring the unit listing page so we can determine this based on content instead (like the specialist unit listing page does)

File: src/node-lib/curriculum-api-2023/fixtures/unitListing.fixture.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { UnitListingPageData } from "../queries/unitListing/unitListing.schema";
+
+const lessonListingFixture = (
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/fixtures/unitListing.fixture.ts:3]: `unitListingFixture` ?

File: src/node-lib/curriculum-api-2023/helpers/index.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+export const filterOutCoreTier = (
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/helpers/index.ts:9]: can we add a comment explaining why we need to do this

**Comment by njblore** [comment]
the order of the tabs is different now, not sure if correct or incorrect!

**Comment by k-huggs** [comment]
> the order of the tabs is different now, not sure if correct or incorrect! 

This is now the correct order I believe

Screenshot 2024-04-18 at 11 33 54

File: src/node-lib/curriculum-api-2023/helpers/index.ts (hunk 5 above)
Comment (k-huggs) [reply src/node-lib/curriculum-api-2023/helpers/index.ts:10]: ✅

**Review by jack-skerrett-oak — APPROVED** [review]
I'll need to raise a ticket for the "no theme" disappearance, but let's get this in.

**Pull request merged**